core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -705,7 +706,8 @@ public void setHeaders(Map<String, List<String>> additionalHttpHeaders) {
 
         @Override
         public Map<String, List<String>> headers() {
-            return CollectionUtils.unmodifiableMapOfLists(config.option(ADDITIONAL_HTTP_HEADERS));
+            Map<String, List<String>> option = Validate.getOrDefault(config.option(ADDITIONAL_HTTP_HEADERS), HashMap::new);
```

Comment:
```suggestion
            Map<String, List<String>> option = Validate.getOrDefault(config.option(ADDITIONAL_HTTP_HEADERS), Collections::emptyMap);
```

No need to create a new object, we can use the statically defined default one. 

Can we also add a `Validate.paramNotNull(⋯)` [here](https://github.com/aws/aws-sdk-java-v2/blob/e5570f717085d451782c0c1ea0006f8ac35a15b3/utils/src/main/java/software/amazon/awssdk/utils/UnmodifiableMapOfLists.java#L43)?